x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticator.java

```diff
@@ -114,9 +108,35 @@ public void close() {
         jwtSignatureValidator.close();
     }
 
+    public JwtRealmSettings.TokenType getTokenType() {
+        return tokenType;
+    }
+
     // Package private for testing
     JwtSignatureValidator.DelegatingJwtSignatureValidator getJwtSignatureValidator() {
         assert jwtSignatureValidator instanceof JwtSignatureValidator.DelegatingJwtSignatureValidator;
         return (JwtSignatureValidator.DelegatingJwtSignatureValidator) jwtSignatureValidator;
     }
+
+    private static List<JwtFieldValidator> configureFieldValidatorsForIdToken(RealmConfig realmConfig) {
+        assert realmConfig.getSetting(JwtRealmSettings.TOKEN_TYPE) == JwtRealmSettings.TokenType.ID_TOKEN;
+        final TimeValue allowedClockSkew = realmConfig.getSetting(JwtRealmSettings.ALLOWED_CLOCK_SKEW);
+        final Clock clock = Clock.systemUTC();
+        return List.of(
+            JwtTypeValidator.INSTANCE,
+            // TODO: mandate "sub" claim once access token support is in place
```

Comment:
> // TODO: mandate "sub" claim once access token support is in place

Will that be passive ?

Reply:
No. The JWT realm doesn't currently mandate `sub` claim because it needs to be lenient to allow some access tokens. This lenience will be removed id_token support once we have dedicated access token support in place.